Add archive in Vault. Closes #180. Closes #181. Closes #226.

juliangruber opened the pull request:
Closes #180. Closes #181. Closes #226.

Updated the vault welcome modal:

![image](https://user-images.githubusercontent.com/10247/91854367-ed936a00-ec63-11ea-998a-46160e53011a.png)

Removed the back arrow from the last slide, so the user can't get back and needs to choose again:

![image](https://user-images.githubusercontent.com/10247/91854380-f2581e00-ec63-11ea-8f0d-db197649108b.png)

Disable the vault again:

![image](https://user-images.githubusercontent.com/10247/91854397-f8e69580-ec63-11ea-8d7b-100f3f7b1dfc.png)

Enable it at any time:

![image](https://user-images.githubusercontent.com/10247/91854411-fedc7680-ec63-11ea-921a-2ccb2c1aecd5.png)

jameslibscie reviewed (changes requested):
The vault seems to get enabled (judging by the menu bar text) regardless of whether I click "Enable the Vault" or "Skip for now..."

jameslibscie commented:
Currently on an unstable connection and got this error:
![image](https://user-images.githubusercontent.com/59870484/91860746-567ee000-ec6c-11ea-8697-654dfc925e0c.png)

I'm also wondering, how do we handle it if someone adds content to their profile while their internet connection is down? It still needs to find its way to the vault somehow.

juliangruber commented:
We can keep retrying, yes, also across restarts.

Do you think this is something that would need to be exposed in the UI

juliangruber commented:
Good point to check for offline breakage..

juliangruber commented:
I fixed the vault always being enabled and also added a 1s loop that will try to upload any previously failed modules, as well es not throwing an error when uploading failed.

chartgerink reviewed (changes requested):
This is looking awesome 😄 I really like how frictionless the Vault integrates. 

My only main question is: Would the Vault need to be disabled upon reset of the database? The Hypergraph Vault stays enabled at the moment (no problem) and doesn't disable when I use the "Skip for now" option. I would expect the Vault to be disabled when I don't click "Enable Vault". What do you think?

Also double checked the offline issue, and all works well on my end 😊 

EDIT: My suggestions are minor, the only reason I didn't approve is the question above.

jameslibscie commented on app/components/welcome/welcome.js:
```diff
@@ -182,10 +191,26 @@ const dialogs = [
+        >
+          <ButtonGroup>
+            <Button emphasis='top' autoFocus>
+              Enable the Vault
```
Dropbox now too! I think as long as we introduce it as Hypergraph Vault it's OK and clear that we're referring to our own vault, but it can't hurt to reiterate at key moments.

psobrakseaton replied:
+1 to this.

jameslibscie reviewed (commented):
Works great!
I'm wondering though about Chris's comment about resetting the DB and then also the re-open welcome modals issue as they both allow the user to make a choice again. Should "Skip for now..." then disable the vault? I'm leaning towards yes, and then perhaps replacing this button with a "Disable" button later on if the setting is already present (or just implementing that now?). Kinda unlikely scenarios early on, but I think it's quite a key part of the software and something that we should be clear on. Would be a shame to lose some users' trust over this.

juliangruber commented on app/lib/vault.js:
```diff
@@ -0,0 +1,62 @@
+
+const addToQueue = url => {
+  const queue = store.get('queue', [])
+  store.set('queue', [...queue, url])
```
> As I read the code (might well be wrong) this is doing an unauthenticated POST request to the server; is this correct?

This is correct.

> Is it possible for people to feed the Hypergraph Vault random Hyper(graph) keys outside of this environment, if they were to inspect the code or their packets?

Yes, they could figure this out either by reading Desktop's, CLI's or the source code of the Hypergraph Vault itself.

There's no way to add a "secret" to Hypergraph Desktop that can't be discovered, the only way to secure this is by requiring the user to log in with the Vault, for which we don't have the UI yet.

I think this is good for now, at our size an attack is a good indicator, if anything.

juliangruber commented:
> Would the Vault need to be disabled upon reset of the database?

I don't think so, since the database is a p2pcommons feature, while the vault is not directly related to it. In this sense, clearing the database means resetting the data, but _not_ the app settings.

I fixed the "Skip for now..." button not disabling the vault. And when you reset the database, you'll be prompted for the vault setting anyway.

juliangruber commented:
via @chartgerink:

![image](https://user-images.githubusercontent.com/10247/92095358-d8d6e380-edd5-11ea-8333-f35cc6e20e28.png)

juliangruber commented:
> via @chartgerink:

This has been fixed in main :)

chartgerink reviewed (approved):
Yay all works 🎉 

My only minor comment that is not blocking merge: When I enable vault in the welcome modals, the menu bar doesn't switch until I restart the application.

juliangruber commented:
> When I enable vault in the welcome modals, the menu bar doesn't switch until I restart the application.

Oh dang this works on Windows, I'll, have to retest

chartgerink reviewed (approved):
![](https://media.giphy.com/media/dXjaXGjVdExGuisFFg/giphy.gif)
